web/client-api/src/main/java/io/deephaven/web/client/api/barrage/BarrageUtils.java

```diff
@@ -182,54 +177,46 @@ private static ColumnData readArrowBuffer(ByteBuffer data, Iter<BarrageFieldNode
         switch (columnType) {
             // for simple well-supported typedarray types, wrap and return
             case "int":
-                assert positions.length().toFloat64() - positions.offset().toFloat64() >= size * 4;
-                Int32Array intArray = new Int32Array(TypedArrayHelper.unwrap(data).buffer, (int) (data.position() + positions.offset().toFloat64()), size);
-                data.position((int) (data.position() + positions.length().toFloat64() - positions.offset().toFloat64()));
+                assert positions.length().toFloat64() >= size * 4;
```

Comment:
Can someone educate me about why we would get length or offset as a 64-bit float?

Reply:
Because JS doesn't have non-floats - this toFloat64() is actually a JS method on a JS object that we are calling from JS. It is on a `Long` object, which is flatbuffer.js's int64 emulation, and they provide this as a method for js to read the value as a plain number. This could lose precision, but realistically won't (how big would a buffer need to be to use more than 54 bits...).

So, we ask the js library to give us a "float 64" of the int 64 - and we cast to int32, since we can't actually index into a buffer longer than that anyway.